src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -368,6 +380,43 @@ public function deleteAction(Request $request)
      *
      * @return RedirectResponse
      */
+    public function applyStateMachineTransitionAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::UPDATE);
+        $resource = $this->findOr404($configuration);
+
+        $event = $this->eventDispatcher->dispatchPreEvent(ResourceActions::UPDATE, $configuration, $resource);
+
+        if ($event->isStopped() && !$configuration->isHtmlRequest()) {
+            throw new HttpException($event->getErrorCode(), $event->getMessage());
+        }
+        if ($event->isStopped()) {
+            $this->flashHelper->addFlashFromEvent($configuration, $event);
+
+            return $this->redirectHandler->redirectToResource($configuration, $resource);
+        }
+
+        $this->stateMachine->apply($configuration, $resource);
+        $this->manager->flush();
+
+        $this->eventDispatcher->dispatchPostEvent(ResourceActions::UPDATE, $configuration, $resource);
+
+        if (!$configuration->isHtmlRequest()) {
+            return $this->viewHandler->handle($configuration, View::create(null, 204));
```

Comment:
Amm, I think should return `200` status and response `resource` to get new state of resource, The client can't know about what's new state changed to.